### PR TITLE
SPUI-533/516/517 > Wallet screen UI

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/app/shared/features/market_details/view/components/balance_block/components/balance_frame.dart [hunk 1]
```diff
@@ -17,24 +17,21 @@ class BalanceFrame extends StatelessWidget {
...
-          padding: EdgeInsets.symmetric(
-            horizontal: 24.w,
-            vertical: 24.h,
-          ),
-          child: child,
+      ),
+      child: Padding(
+        padding: EdgeInsets.symmetric(
+          horizontal: 24.w,
+          vertical: 24.h,
```
Comment (eli1stark) [inline lib/app/shared/features/market_details/view/components/balance_block/components/balance_frame.dart:32]: color and scalability

File: lib/app/shared/features/wallet/view/components/action_button.dart [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
-      backgroundColor: Colors.grey.shade200,
-      height: 104.h,
-      child: AppButtonWhite(
-        name: 'Action',
-        onTap: () {
-          // TODO(any): Add action bottom sheet
-        },
-      ),
+    final currencies = useProvider(currenciesPod);
+    final isNotEmptyBalance = !isBalanceEmpty(currencies);
```
Comment (eli1stark) [inline lib/app/shared/features/wallet/view/components/action_button.dart:16]: unused variables

File: lib/app/shared/features/wallet/view/components/wallets_body/components/card_block/components/wallet_card.dart [hunk 3]
```diff
@@ -27,81 +27,93 @@ class WalletCard extends HookWidget {
...
+    var cardColor = SColorsLight().greenLight;
 
-    return Expanded(
-      child: Container(
-        height: 0.25.sh,
-        margin: EdgeInsets.symmetric(
-          horizontal: 24.w,
+    if (marketItem.dayPercentChange.isNegative) {
+      buttonColor = SColorsLight().red;
+      cardColor = SColorsLight().redLight;
```
Comment (eli1stark) [inline lib/app/shared/features/wallet/view/components/wallets_body/components/card_block/components/wallet_card.dart:36]: not simple color

File: lib/app/shared/features/wallet/view/components/wallets_body/components/transactions_list_item/transaction_list_item.dart [hunk 4]
```diff
@@ -24,6 +24,11 @@ class TransactionListItem extends HookWidget {
   @override
   Widget build(BuildContext context) {
     final hidden = useProvider(walletHiddenStPod);
+    var color = SColorsLight().green;
+
+    if (transactionListItem.balanceChange.isNegative) {
+      color = SColorsLight().red;
+    }
```
Comment (eli1stark) [inline lib/app/shared/features/wallet/view/components/wallets_body/components/transactions_list_item/transaction_list_item.dart:31]: not simple color

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/shared/features/market_details/view/components/balance_block/components/balance_frame.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/market_details/view/components/balance_block/components/balance_frame.dart:32]: file is deleted

File: lib/app/shared/features/wallet/view/components/action_button.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/action_button.dart]: deleted

File: lib/app/shared/features/wallet/view/components/wallets_body/components/card_block/components/wallet_card.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallets_body/components/card_block/components/wallet_card.dart:34]: done

File: lib/app/shared/features/wallet/view/components/wallets_body/components/transactions_list_item/transaction_list_item.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallets_body/components/transactions_list_item/transaction_list_item.dart:31]: done

**Pull request merged**